src/Tus/Client.php

```diff
@@ -172,6 +175,79 @@ public function getChecksum() : string
         return $this->checksum;
     }
 
+    /**
+     * Add metadata.
+     *
+     * @param string $key
+     * @param string $value
+     *
+     * @return Client
+     */
+    public function addMetadata(string $key, string $value) : self
+    {
+        $this->metadata[$key] = base64_encode($value);
+
+        return $this;
+    }
+
+    /**
+     * Remove metadata.
+     *
+     * @param string $key
+     *
+     * @return Client
+     */
+    public function removeMetadata(string $key) : self
+    {
+        unset($this->metadata[$key]);
+
+        return $this;
+    }
+
+    /**
+     * Set metadata.
+     *
+     * @param array $items
+     *
+     * @return Client
+     */
+    public function setMetadata(array $items) : self
+    {
+        $items = array_map('base64_encode', $items);
+
+        $this->metadata = $items;
+
+        return $this;
+    }
+
+    /**
+     * Get metadata.
+     *
+     * @return array
+     */
+    public function getMetadata() : array
+    {
+        $this->addMetadata('filename', $this->fileName);
```

Comment:
why do we need to add metadata inside `getMetadata` method?

Reply:
For insure metadata always has `filename`, but it can check if `filename` not exist then add.

Reply:
IMO It should not be `getMetadata` method that would add a `filename` meta. It should simply return metadata. I think it is more meaningful to add that filename meta as soon as we get the filename. Somewhere around [here](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L87-L89).

Reply:
It sounds better, however it just set in [here](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L87-L89) or also set it in [here](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L112-L113)?

Reply:
I agree with @ankitpokhrel `getMetaData` should only return metadata and should not make any additional changes to internal data.

Like mentioned before, we can add `filename` to metadata if it is needed afterwards in some parts of the application. Perhaps, as mentioned [this](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L87-L89) could be suitable place. Or better if we have `setMetadata` we can add `filename` there.

Reply:
It is better to can override `filename` with `setMetadata` or `addMetadata`, so [this](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L87-L89) is better place I think.